README.md
<h1 align="center">ePSXe64Ubuntu</h1>

<p align="center">Version 11.6 | <a href="https://github.com/brandleesee/ePSXe64Ubuntu/blob/master/CHANGELOG.md">Changelog</a> | <a href="https://chrismartinphotography.files.wordpress.com/2013/01/2012-landscapes-c2a9-christopher-martin-93441.jpg">Background</a> | <a href="http://xenlism.github.io/minimalism/">Minimalism GTK3 Theme</a> | <a href="https://github.com/xenlism/Storm">Storm Icon Set</a> | <a href="https://b00merang.weebly.com/windows-10.html">Windows 10 GTK Pack</a></p>

<img src="https://i.imgur.com/2DlxMw4.png" />

<h2 align="center">overview</h2>

<p align="center">an interactive script that installs ePSXe 2.0.5 Linux x64 on Debian, Ubuntu and other x64 derivatives</p>

<h2 align="center">features</h2>

- [x] uses built-in BIOS HLE
- [x] uses built-in Core GPU & SPU plugins
- [x] installs specific ePSXe dependency: `libsdl-ttf2.0-0`
- [x] installs decompression tool: `ecm`
- [x] installs Dash / Dock / Panel application icon
- [x] creates `.desktop` by default for launcher integration
- [x] lock-to-launcher instructions
- [x] interactive Shaders back-up function
- [x] `libcurl 3` & `libcurl 4` now co-exist enabling software like Kodi etc to work simultaneously with ePSXe
- [ ] test & add more shaders ([bundle to test](https://www.ngemu.com/threads/links-and-guides-to-custom-shaders-for-petes-opengl2-plugin.76098/page-51#post-2722008))

<h2 align="center">typical installation</h2>

<img src="https://i.imgur.com/jJyFe8b.gif" />

<h2 align="center">installation procedure</h2>

<p align="center">in terminal</p>

```bash
wget -O ePSXe64Ubuntu.sh https://raw.githubusercontent.com/brandleesee/ePSXe64Ubuntu/master/ePSXe64Ubuntu.sh && bash ePSXe64Ubuntu.sh
```

<img src="https://i.imgur.com/CD9XlIQ.png" />

<p align="center">enter <strong>SUDO</strong> password</p>

<p align="center"><strong>RIGHT CLICK</strong> on ePSXe icon in Dash / Dock / Panel to Add to Favorities / Lock</p>

<p align="center"><img src="https://i.imgur.com/8rOUO7A.png" /></p>

<p align="center"><strong>CLOSE</strong> ePSXe window to continue with script.</p>

<img src="https://i.imgur.com/MZMI4rv.png" />

<p align="center"><strong>download</strong>, <strong>restore</strong> or <strong>ignore</strong> bundled shaders

<p align="center">type number and press <strong>ENTER</strong></p>

<p align="center"><img src="https://i.imgur.com/tmswgZJ.png" /></p>

<h2 align="center">decompressing using ecm</h2>

```bash
cd 'INSERT LOCATION of COMPRESSED FILE'

ecm-uncompress 'long name of game.bin.ecm'
```

<h2 align="center">normal .cue sheets</h2>

```
FILE "_____ ___ _______.bin" BINARY
  TRACK 01 MODE2/2352
    INDEX 01 00:00:00
```

* `  .bin  ` and `  .cue  ` must be in the same folder.
* `  .cue  ` must be named exactly as `  .bin  ` (only difference being the file's extension).
* The first line within the `  .cue  ` sheet must contain the **exact** name of the `  .bin  ` file inside double quotes including the `  .bin  ` extension.
* Some games require elaborate `  .cue  ` sheets. A Google search should see you sorted.

<h2 align="center">removing ePSXe</h2>

```bash
sudo rm /usr/share/applications/ePSXe.desktop
```
If you are running Ubuntu 18.04 or later:
```bash
sudo rm /usr/lib/x86_64-linux-gnu/libcurl.so.3
```

<p align="center"><strong>SUDO</strong> password required</p>

<p align="center"><strong>DELETE</strong>:</p>

<p align="center">ePSXe | ePSXe64Ubuntu.sh | ePSXe_backups | .epsxe | .epsxe.svg</p>

<p align="center" ><strong>restart</strong> pc</p>

<img src="https://i.imgur.com/FiBbK8w.png" />


CHANGELOG.md
<h1 align="center">ePSXe64Ubuntu</h1>

<table>
<tbody>
  <tr>
   <td><img src="https://camo.githubusercontent.com/66f9d39d7a5274a0c8e7e0e5ff353eba364160c2/687474703a2f2f692e696d6775722e636f6d2f5534514e6b584c2e706e67" /></td>
  </tr>
</tbody>
</table>

<h2 align="center">changelog</h2>

<h3 align="center">11</h3>

<table>
<tbody>
  <tr>
   <td><img src="https://user-images.githubusercontent.com/12674437/52143698-6bdcf400-265c-11e9-99dd-79090db2bb39.png" /></td>
  </tr>
</tbody>
</table>

- 11.6
  - Now using Ubuntu's archive for necessary packages.
  - Checksum is implemented.
  - Shellcheck used to correct common errors.

- 11.5
  - Now script allows for multiple users - https://github.com/brandleesee/ePSXe64Ubuntu/pull/23.

- 11.4
  - Simplify libcurl4 conflict detection by adding generic code that looks for the package instead of adding specific lines per affected distribution - https://pkgs.org/download/libcurl4
  - ecm no longer in repositories, thus, updated to other mirror

- 11.3
  - Added support for Debian Sid testing branch, Debian 9 and Debian 10 and KDE neon. This because libcurl 4 was replacing libcurl 3 on updating - https://github.com/brandleesee/ePSXe64Ubuntu/issues/7.
  - https://github.com/brandleesee/ePSXe64Ubuntu/commit/5f3b1bda3873336591b151178a90400ae2b11212

- 11.2
  - Consolidated code blocks as per this comment: https://github.com/brandleesee/ePSXe64Ubuntu/commit/68a031d9800b308ebdc9f55046dc7f2286267e7c#commitcomment-32485422

- 11.1
  - Linux Mint: apt-get Legacy Utility Command
    - Linux Mint still uses apt-get out-of-the-box instead of Apt.
	- issue: https://github.com/cromat/ePSXe64Ubuntu/commit/9deb4bc17ea5e53d30c6023faca060ac612be6e4
  - Libcurl 3 - 4 issue on Linux Mint 19 (https://forums.linuxmint.com/viewtopic.php?t=274477)

  - https://ryanstutorials.net/bash-scripting-tutorial/bash-if-statements.php - to make sure *elif* statement is used correctly
  - https://forums.linuxmint.com/viewtopic.php?t=263277 - checked where packages are downloaded from - resulted to be from ubuntu
  - https://www.rootusers.com/check-which-linux-mint-version/
  - https://www.tldp.org/LDP/abs/html/comparison-ops.html
  - https://stackoverflow.com/a/4277753
  - https://stackoverflow.com/questions/2237080/how-to-compare-strings-in-bash

- 11
  - easier install (one-liner in terminal kicks up the script)
  - removed quiet mode so as to have visible any errors while installing dependencies
  - fixes the problem where recent versions of ubuntu can't have libcurl3 installed at the same time as libcurl4 or in the case of 18.10, it's not available at all. It does this by downloading the 18.04 version of libcurl3 and installing it with the name libcurl.so.3 instead of libcurl.so.4. It then patches the ePSXe executable to look for libcurl.so.3 instead of libcurl.so.4.
  - Libcurl 3 is installed alongside libcurl4 so things like kodi should work regardless of whether the checksum matches or not.
  - patch: https://github.com/brandleesee/ePSXe64Ubuntu/pull/5
  - issue: http://ngemu.com/threads/ubuntu-bionic-libcurl-so-4-curl_openssl_3-not-found.204577/
  - previous workaround: https://launchpad.net/~xapienz/+archive/ubuntu/curl34

<h3 align="center">10</h3>

- 10.3
  - Fixed script running from git cloned repo.
  - Fixed script running from Steam's Big Picture Mode.

- 10.2
  - Added Debian support by adding libssl1.0.0.

- 10.1
  - Renamed icon to become hidden by default.
  - Fixed removal of icon. Consequence of v10's restructuring.

- 10
  - Optimized for Ubuntu 18.04.
    - Prior to 18.04 libcurl3 was installed only for Xubuntu / xfce.
	  - Now installed by default.
	  - http://ngemu.com/threads/ubuntu-bionic-libcurl-so-4-curl_openssl_3-not-found.204577/
	- Restructured script to first download icon then create the .desktop file effectively removing any instances of a blank Dash/Dock/Panel placeholder on first install.

<h3 align="center">9</h3>

<table>
<tbody>
  <tr>
   <td align="center"><img src="http://i.imgur.com/cSqEyXC.gif" /></td>
  </tr>
</tbody>
</table>

- 9.2
  - Changed text colour output.
  - Wrote clearer GUI instructions.
  - Removed most titles as they were misleading and did not promulgate useful information.
  - Commented sections and skipped lines for easier reference.

- 9.1
  - Corrected 1st line. Missing `/`.

- 9
  - Major revision.
  - Proper back up functionality.
  - Interactive SHADERS menu.
  - Prints terminal entries in color.
  - Installs required packages silently.
  - Moves and copies silently.
  - Makes effective use of system `/tmp`.
  - Added disclaimer.
  - Credit given.

<h3 align="center">8</h3>

<table>
<tbody>
  <tr>
   <td align="center"><img src="http://i.imgur.com/kfMSU17.png" /></td>
   <td align="center"><img src="http://i.imgur.com/W1zggGH.gif" /></td>
  </tr>
</tbody>
</table>

- 8.1
  - Proper branding of ePSXe64Ubuntu.
  - Added ecm tools.

- 8
  - Partial backup function.

<h3 align="center">7</h3>

<table>
<tbody>
  <tr>
   <td align="center"><img src="http://i.imgur.com/2Bw3iLS.png" /></td>
   <td align="center"><img src="http://i.imgur.com/fRardY8.gif" /></td>
  </tr>
</tbody>
</table>

- 7.3
  - Added `libcurl3` for Xubuntu's xfce.

- 7.2
  - Changed `apt` to `apt-get` for better support on Linux Mint 13.3.

- 7.1
  - Removed redundant line.

- 7
  - Added Icon for Dash / Dock / Panel.
  - General Re-write with if/then statements.

<h3 align="center">6</h3>

<table>
<tbody>
  <tr>
   <td align="center"><img src="http://i.imgur.com/2mT7smo.png" /></td>
   <td align="center"><img src="http://i.imgur.com/ZBFCUva.png" /></td>
  </tr>
  <tr>
   <td align="center"><img src="http://i.imgur.com/57kdpJR.png" /></td>

e64u.sh
#!BIN/BASH

# ePSXe: emulator is property of ePSXe team
# ePSXe64Ubuntu: script is property of Brandon Camilleri ( brandleesee / Yrvyne )
# contact: brandon.camilleri.90@gmail.com
# repository: https://github.com/brandleesee/ePSXe64Ubuntu

ins="ePSXe205linux_x64.zip"
tmp="/home/$USER/ePSXe"
hid="/home/$USER/.epsxe"
bkp="/home/$USER/ePSXe_backup/$(date '+%Y_%m_%d_%R')"
cor="/usr/share/applications"

echo
echo "*************"
echo "Prerequisites"
echo "*************"
echo
echo "  <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<"
echo "  INTERACTIVITY 1 : SUDO password required"
echo "  >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
echo
		sudo apt-get -y install libsdl-ttf2.0-0 libcurl3 unzip
echo
echo "******"
echo "Backup"
echo "******"
echo
echo "  ===================================="
echo "  Move 'old' epsxe to backup directory"
echo "  ===================================="
echo
		if [ -d $hid ]; then
		mkdir -p $bkp
		mv $hid $bkp
		fi
echo
echo "***********************"
echo "Remove any Duplicate..."
echo "***********************"
echo
echo "  ============================="
echo "  ...icon in share/applications"
echo "  ============================="
echo
		if [ -e $cor/epsxe.desktop ]; then
		sudo rm -rf $cor/epsxe.desktop
		fi
echo
echo "  ================================"
echo "  ...ePSXe directory & recreate it"
echo "  ================================"
echo
		if [ -d $tmp ]; then
		rm -rf $tmp
		mkdir -p $tmp
		fi
echo
echo "****"
echo "Icon"
echo "****"
echo
echo "  ============="
echo "  Download Icon"
echo "  ============="
echo
		wget https://github.com/brandleesee/ePSXe64Ubuntu/raw/master/epsxe.svg -P $hid
echo
echo "  ==============="
echo "  Create .desktop"
echo "  ==============="
echo
		echo "[Desktop Entry]" > "$hid/epsxe.desktop"
		echo "Type=Application" >> "$hid/epsxe.desktop"
		echo "Terminal=false" >> "$hid/epsxe.desktop"
		echo "Exec=./epsxe" >> "$hid/epsxe.desktop"
		echo "Name=ePSXe" >> "$hid/epsxe.desktop"
		echo "Comment=Created using ePSXe64Ubuntu script from https://github.com/brandleesee" >> "$hid/epsxe.desktop"
		echo "Icon=$hid/epsxe.svg" >> "$hid/epsxe.desktop"
		echo "Categories=Game;Emulator;" >> "$hid/epsxe.desktop"
echo
echo "  ================"
echo "  Transfer.desktop"
echo "  ================"
echo
		sudo mv $hid/epsxe.desktop $cor
echo
echo "*****"
echo "ePSXe"
echo "*****"
echo
echo "  <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<"
echo "  INTERACTIVITY 2 : LOCK TO LAUNCHER"
echo "                    Now is the ideal time to add to favorities:" 
echo "                    Right click on icon in Dash/Dock"
echo "                    Select Add to Favorities / Lock to Launcher"
echo "  >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
echo
echo "  <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<"
echo "  INTERACTIVITY 3 : ePSXe has opened its GUI."
echo "                    CLOSE it to continue." 
echo "                    This creates the hidden .epsxe folder"
echo "                    in $USER."
echo "  >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
echo
echo "  =============="
echo "  Download ePSXe"
echo "  =============="
echo
		wget http://www.epsxe.com/files/$ins -P $tmp
echo
echo "  ============="
echo "  Extract ePSXe"
echo "  ============="
echo
		unzip $tmp/$ins -d $tmp
echo
echo "  ================================="
echo "  Move executable to Home directory"
echo "  ================================="
echo
		mv $tmp/epsxe_x64 /home/$USER/epsxe
echo
echo "  ====================="
echo "  Make ePSXe executable"
echo "  ====================="
echo
		sudo chmod +x epsxe
echo
echo "  =========="
echo "  Open ePSXe"
echo "  =========="
echo
		./epsxe
echo
echo "  =============================="
echo "  Transfer docs folder to .epsxe"
echo "  =============================="
echo
		mv $tmp/docs $hid
echo
echo "  ================="
echo "  Activate BIOS HLE"
echo "  ================="
echo
		sed -i '11s/.*/BiosPath = /' $hid/epsxerc
		sed -i '14s/.*/BiosHLE = 1/' $hid/epsxerc
echo
echo "  *******"
echo "  Shaders"
echo "  *******"
echo
echo "    ================"
echo "    Download shaders"
echo "    ================"
echo
		wget https://github.com/brandleesee/ePSXe64Ubuntu/raw/master/shaders.zip -P $tmp
echo
echo "    ==============="
echo "    Extract shaders"
echo "    ==============="
echo
		unzip $tmp/shaders.zip -d $hid/shaders
echo
echo "  ============================"
echo "  Removing Temporary Directory"
echo "  ============================"
echo
		rm -rf $tmp
echo
echo "****************"
echo "Restoring Backup"
echo "****************"
echo
		cp -avr $bkp/.epsxe/cheats/. $hid/cheats
		cp -avr $bkp/.epsxe/config/. $hid/config
		cp -avr $bkp/.epsxe/configs/. $hid/configs
		cp -avr $bkp/.epsxe/covers/. $hid/covers
		cp -avr $bkp/.epsxe/idx/. $hid/idx
		cp -avr $bkp/.epsxe/info/. $hid/info
		cp -avr $bkp/.epsxe/memcards/. $hid/memcards
		cp -avr $bkp/.epsxe/patches/. $hid/patches
		cp -avr $bkp/.epsxe/plugins/. $hid/plugins
		cp -avr $bkp/.epsxe/shaders/. $hid/shaders
		cp -avr $bkp/.epsxe/sstates/. $hid/sstates
echo
echo "***************"
echo "Script Finished"
echo "***************"
echo
echo "  <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<"
echo "  For more information visit:"
echo "  https://github.com/brandleesee/ePSXe64Ubuntu"
echo "  >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
echo


siblings/shaders-only.sh
#!BIN/BASH

echo
echo "**************"
echo "Pre-Requisites"
echo "**************"
echo
echo "  <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<"
echo "  INTERACTIVITY 1 : SUDO password required"
echo "  >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
echo
		sudo apt -y install zip unzip
echo
echo "  ============================"
echo "  Create Temporary Directories"
echo "  ============================"
echo
		mkdir /tmp/ePSXe
echo
echo "  *******"
echo "  Shaders"
echo "  *******"
echo
echo "    ================"
echo "    Download shaders"
echo "    ================"
echo
		wget http://www.epsxe.com/files/shaders.zip -P /tmp/ePSXe
echo
echo "    ==============="
echo "    Extract shaders"
echo "    ==============="
echo
		unzip /tmp/ePSXe/shaders.zip -d ~/.epsxe/shaders
echo
echo
echo "  =============================="
echo "  Removing Temporary Directories"
echo "  =============================="
echo
		rm -rf /tmp/ePSXe
		rm -rf ~/.epsxe/shaders/shaders
echo
echo "*****************"
echo "Shaders Installed"
echo "*****************"
echo
		cd ~
echo



ePSXe64Ubuntu.sh
#!/bin/bash

# ePSXe emulator is property of ePSXe team, http://epsxe.com/, under Proprietary license.
# ePSXe64Ubuntu.sh and formerly e64u.sh scripts are property of Brandon Lee Camilleri ( blc / brandleesee / Yrvyne , https://twitter.com/brandleesee , https://www.reddit.com/user/Yrvyne/ )
# ePSXe64Ubuntu.sh and formerly e64u.sh scripts are protected under the vestiges of GNU GENERAL PUBLIC LICENSE Version 3, 29 June 2007.
# Disclaimer: Brandon Lee Camilleri ( blc / brandleesee / Yrvyne ) does not assume any responsibilities and shall not be held liable should ePSXe64Ubuntu.sh, e64u.sh, shaders.zip, .ePSXe.svg CHANGELOG.md and/or README.md fail in their intended purpose, attempt and usage and/or break the system/s being used on.
# Brandon Lee Camilleri ( blc / brandleesee / Yrvyne ) can be reached on brandon.camilleri.90@gmail.com
# ePSXe64Ubuntu repository can be found at https://github.com/brandleesee/ePSXe64Ubuntu

# Leave anything with ~ unquoted so it expands properly.  This lets us handle complicated home directory locations
ver="11.6"
ins="ePSXe205linux_x64.zip"
hme=~
hid=~/.epsxe
bkp=~/ePSXe_backups/$(date "+%F-%T-%Z")
cor=~/.local/share/applications
exe=~/.local/bin/ePSXe
dls="https://raw.githubusercontent.com/brandleesee/ePSXe64Ubuntu/master"
opt=("Download" "Restore from backup")
PS3="Choose from 1 to 3 above. "
PROTO="http"
MIRROR="archive.ubuntu.com"

check_sha256sum() {
	tempfile=$1
	filehash=$2
	if [ ! -f "${tempfile}" ]; then
		tput setaf 1; echo "  ERROR: File ${tempfile} doesn't exist"; tput sgr0
		exit 1
	elif [ "$(sha256sum "${tempfile}" | head -c 64)" = "${filehash}" ]; then
		tput setaf 2; echo "${tempfile} matches provided sha256sum"; tput sgr0
	else
		tput setaf 1; echo "  ERROR: ${tempfile} doesn't match provided sha256sum"; tput sgr0
		exit 1
	fi
}

tput setaf 2; echo "Welcome to ePSXe64Ubuntu.sh script, $ver."; tput sgr0
tput setaf 1; echo "When ePSXe window appears on screen:"; tput sgr0
tput setaf 1; echo "  Right click on icon in Dash/Dock/Panel"; tput sgr0
tput setaf 1; echo "  Add to Favorites/Lock"; tput sgr0
tput setaf 1; echo "  CLOSE ePSXe GUI to continue with the script."; tput sgr0
tput setaf 2; echo "Script started."; tput sgr0

sudo apt-get update
sudo apt-get -y install wget sed
# xxd was provided by vim-common on older distros
sudo apt-get -y install xxd || sudo apt-get -y install vim-common

# Install ubuntu 18.04 version of openssl1.0.0 if it's not known to our version of our distribution
if ! apt-cache show libssl1.0.0 2>/dev/null|grep -q '^Package: libssl1.0.0$'
then
	filename="libssl1.0.0_1.0.2n-1ubuntu5_amd64.deb"
	tempfile="/tmp/${filename}"
	filehash="fcadc659174561b7a925e4f17e9de7451f4fb556a032fea1ed2ff800ed3a285e"
	wget "${PROTO}://${MIRROR}/ubuntu/pool/main/o/openssl1.0/${filename}" -O "${tempfile}"
	check_sha256sum "${tempfile}" "${filehash}"
	sudo dpkg --force-depends -i "${tempfile}"
	sudo apt-get -y install -f
	rm "${tempfile}"
fi

if ! apt-cache show ecm 2>/dev/null|grep -q '^Package: ecm$'
then
	filename="ecm_1.03-1build1_amd64.deb"
	tempfile="/tmp/${filename}"
	filehash="3889b926bcaed64bfc66f20c27f943e63ec41c701d1d50682b21f06f95d6fcfd"
	wget "${PROTO}://${MIRROR}/ubuntu/pool/universe/c/cmdpack/${filename}" -O "${tempfile}"
	check_sha256sum "${tempfile}" "${filehash}"
	sudo dpkg --force-depends -i "${tempfile}"
	sudo apt-get -y install -f
	rm "${tempfile}"
fi

# Installs required packages per OS
if apt-cache show libcurl4 2>/dev/null|grep -q '^Package: libcurl4$'
then
	sudo apt-get -y install libncurses5 libsdl-ttf2.0-0 libssl1.0.0 ecm unzip
	filename="libcurl3_7.58.0-2ubuntu2_amd64.deb"
	tempfile="/tmp/${filename}"
	filehash="26d8e98614a55013b35afac465081ec17c9d931ee11f648bca7c3cbaefb404af"
	wget "${PROTO}://${MIRROR}/ubuntu/pool/main/c/curl3/${filename}" -O "${tempfile}"
	check_sha256sum "${tempfile}" "${filehash}"
	sudo mkdir /tmp/libcurl3
	sudo dpkg-deb -x "${tempfile}" /tmp/libcurl3
	sudo cp -vn /tmp/libcurl3/usr/lib/x86_64-linux-gnu/libcurl.so.4.5.0 /usr/lib/x86_64-linux-gnu/libcurl.so.3
	sudo rm -rf /tmp/libcurl3
	rm "${tempfile}"
else
	sudo apt-get -y install libcurl3 libsdl-ttf2.0-0 libssl1.0.0 ecm unzip
fi

# Back-up function
	if [ -d "$hid" ]; then
	  mkdir -p "$bkp"
	  mv "$hid" "$bkp"
	fi

# Removes duplicate of ePSXe executable
	if [ -e "$exe" ]; then
	  rm -rf "$exe"
	fi

# Downloads Icon
	mkdir -p "$hme/.local/share/ePSXe"
	wget -q "$dls/.ePSXe.svg" -O "$hme/.local/share/ePSXe/ePSXe.svg"

# Checks and creates icon data for Dash/Dock/Panel
	if [ -e "$cor/ePSXe.desktop" ]; then
	  rm -rf "$cor/ePSXe.desktop"
	fi
	echo "[Desktop Entry]" > "/tmp/ePSXe.desktop"
	{
	  echo "Type=Application"
	  echo "Terminal=false"
	  echo "Exec=$exe"
	  echo "Name=ePSXe"
	  echo "Comment=Created using ePSXe64Ubuntu from https://github.com/brandleesee"
	  echo "Icon=$hme/.local/share/ePSXe/ePSXe.svg"
	  echo "Categories=Game;Emulator;"
	} >> "/tmp/ePSXe.desktop"
	mkdir -p "$cor"
	mv "/tmp/ePSXe.desktop" "$cor/ePSXe.desktop"

# Sets up ePSXe
	wget -q "https://www.epsxe.com/files/$ins" -P "/tmp" || wget -q "http://www.epsxe.com/files/$ins" -P "/tmp"
	unzip -qq "/tmp/$ins" -d "/tmp"
	mkdir -p "$(dirname $exe)"
	if apt-cache show libcurl4 2>/dev/null|grep -q '^Package: libcurl4$'
	then
	  xxd /tmp/epsxe_x64 /tmp/epsxe_x64.xxd
	  sed -i '6434c \00019210: 2e73 6f2e 3300 6375 726c 5f65 6173 795f  .so.3.curl_easy_' /tmp/epsxe_x64.xxd
	  xxd -r /tmp/epsxe_x64.xxd "$exe"
	  rm -f /tmp/epsxe_x64.xxd
	  if ! sha256sum -c --quiet <(echo "45fb1ee4cb21a5591de64e1a666e4c3cacb30fcc308f0324dc5b2b57767e18ee  $exe")
	  then
	    tput setaf 1; echo "WARNING: patched $exe did not match checksum, using original executable instead"; tput sgr0
	    cp -f /tmp/epsxe_x64 "$exe"
	  fi
	  rm -f /tmp/epsxe_x64
	else
	  mv "/tmp/epsxe_x64" "$exe"
	fi
	chmod +x "$exe"
	"$exe"

# Transfers docs folder to .epsxe
	mkdir -p "$hid"
	mv "/tmp/docs" "$hid"

# Activates BIOS HLE
	sed -i '11c \BiosPath = ' "$hid/epsxerc"
	sed -i '14c \BiosHLE = 1' "$hid/epsxerc"

# Restores Back-Up
	if [ -d "$bkp/.epsxe" ]; then
	  cp -r "$bkp/.epsxe/bios/." "$hid/bios"
	  cp -r "$bkp/.epsxe/cheats/." "$hid/cheats"
	  cp -r "$bkp/.epsxe/config/." "$hid/config"
	  cp -r "$bkp/.epsxe/configs/." "$hid/configs"
	  cp -r "$bkp/.epsxe/covers/." "$hid/covers"
	  cp -r "$bkp/.epsxe/docs/." "$hid/docs"
	  cp -r "$bkp/.epsxe/idx/." "$hid/idx"
	  cp -r "$bkp/.epsxe/info/." "$hid/info"
	  cp -r "$bkp/.epsxe/memcards/." "$hid/memcards"
	  cp -r "$bkp/.epsxe/patches/." "$hid/patches"
	  cp -r "$bkp/.epsxe/plugins/." "$hid/plugins"
	  cp -r "$bkp/.epsxe/sstates/." "$hid/sstates"
	fi

# Function for Shaders
tput setaf 2; echo "Shaders Menu"; tput sgr0
	select opt in "${opt[@]}" "Do nothing"; do
	  case "$REPLY" in
	    1 )
	      wget -q "$dls/shaders.zip" -P "/tmp"
	      unzip -qq "/tmp/shaders.zip" -d "$hid/shaders"
	      echo "This choice has downloaded shaders from ePSXe64Ubuntu repository.";
	      break
	    ;;
	    2 )
	      cp -r "$bkp/.epsxe/shaders/." "$hid/shaders"
	      break
	    ;;
	    $(( ${#opt[@]}+1 )) ) echo "This choice has left the shaders folder empty."; break;;
	    *) echo "Invalid option. Choose from 1 to 3.";continue;;
	  esac
	done

# Removes clutter
	rm -rf "/tmp/$ins"
	rm -rf "/tmp/shaders.zip"

tput setaf 2; echo "Script finished."; tput sgr0
